example/fine_advection/advection_package.cpp

```diff
@@ -118,50 +118,57 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  pkg->CheckRefinementMesh = CheckRefinementMesh;
   pkg->EstimateTimestepMesh = EstimateTimestep;
   pkg->FillDerivedMesh = FillDerived;
   return pkg;
 }
 
-AmrTag CheckRefinement(MeshBlockData<Real> *rc) {
+void CheckRefinementMesh(MeshData<Real> *md, parthenon::ParArray1D<AmrTag> &amr_tags) {
   std::shared_ptr<StateDescriptor> pkg =
-      rc->GetMeshPointer()->packages.Get("advection_package");
+      md->GetMeshPointer()->packages.Get("advection_package");
   auto do_regular_advection = pkg->Param<bool>("do_regular_advection");
   if (do_regular_advection) {
     // refine on advected, for example.  could also be a derived quantity
-    static auto desc = parthenon::MakePackDescriptor<Conserved::phi>(rc);
-    auto pack = desc.GetPack(rc);
-
-    auto pmb = rc->GetBlockPointer();
-    IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
-    IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
-    IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+    static auto desc = parthenon::MakePackDescriptor<Conserved::phi>(md);
+    auto pack = desc.GetPack(md);
 
-    typename Kokkos::MinMax<Real>::value_type minmax;
-    parthenon::par_reduce(
-        parthenon::loop_pattern_mdrange_tag, PARTHENON_AUTO_LABEL, DevExecSpace(), 0,
-        pack.GetNBlocks() - 1, // Runs from [0, 0] since pack built from MeshBlockData
-        pack.GetLowerBoundHost(0), pack.GetUpperBoundHost(0), kb.s, kb.e, jb.s, jb.e,
-        ib.s, ib.e,
-        KOKKOS_LAMBDA(const int b, const int n, const int k, const int j, const int i,
-                      typename Kokkos::MinMax<Real>::value_type &lminmax) {
-          lminmax.min_val = (pack(b, n, k, j, i) < lminmax.min_val ? pack(b, n, k, j, i)
-                                                                   : lminmax.min_val);
-          lminmax.max_val = (pack(b, n, k, j, i) > lminmax.max_val ? pack(b, n, k, j, i)
-                                                                   : lminmax.max_val);
-        },
-        Kokkos::MinMax<Real>(minmax));
-
-    auto pkg = pmb->packages.Get("advection_package");
     const auto &refine_tol = pkg->Param<Real>("refine_tol");
     const auto &derefine_tol = pkg->Param<Real>("derefine_tol");
 
-    if (minmax.max_val > refine_tol && minmax.min_val < derefine_tol)
-      return AmrTag::refine;
-    if (minmax.max_val < derefine_tol) return AmrTag::derefine;
+    auto ib = md->GetBoundsI(IndexDomain::entire);
+    auto jb = md->GetBoundsJ(IndexDomain::entire);
+    auto kb = md->GetBoundsK(IndexDomain::entire);
+    auto scatter_tags = amr_tags.ToScatterView<Kokkos::Experimental::ScatterMax>();
+    parthenon::par_for_outer(
+        PARTHENON_AUTO_LABEL, 0, 0, 0, pack.GetNBlocks() - 1, 0,
+        pack.GetMaxNumberOfVars() - 1, kb.s, kb.e,
+        KOKKOS_LAMBDA(parthenon::team_mbr_t team_member, const int b, const int n,
+                      const int k) {
+          if (pack.GetIndex(b, Conserved::phi()) < 0) return;
```

Comment:
Is there another way to check if a variable in a sparse pack is allocated, maybe using `n`  similar to `VariablePack::IsAllocated`?

Reply:
`SparsePacks` only pack variables that are allocated, so the standard way to check if a field is allocated is to compare 
```c++
bool is_my_field_allocated = sparse_pack.GetUpperBound(b, my_field_t()) >= sparse_pack.GetLowerBound(b, my_field_t())
```